clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3087,34 +3087,26 @@ static void handleMaxGlobalWorkDimAttr(Sema &S, Decl *D,
   if (D->isInvalidDecl())
     return;
 
-  uint32_t MaxGlobalWorkDim;
-  const Expr *E = Attr.getArgAsExpr(0);
-  if (!checkUInt32Argument(S, Attr, E, MaxGlobalWorkDim, 0,
-                           /*StrictlyUnsigned=*/true))
-    return;
-
-  if (MaxGlobalWorkDim > 3) {
-    S.Diag(Attr.getLoc(), diag::err_intel_attribute_argument_is_not_in_range)
-      << Attr;
-    return;
-  }
+  Expr *E = Attr.getArgAsExpr(0);
```

Comment:
And we probably need to return call to checkUInt32Argument or write a value of attribute to `MaxGlobalWorkDim` variable.

Reply:
We are handling this inside void Sema::addIntelSYCLSingleArgFunctionAttr function (sema.h)